remote_config/integration_test/src/integration_test.cc

```diff
@@ -302,6 +303,7 @@ TEST_F(FirebaseRemoteConfigTest, TestAddOnConfigUpdateListener) {
          firebase::remote_config::RemoteConfigError) {});
 #else
   auto config_update_promise = std::make_shared<std::promise<void> >();
+  auto config_update_future = config_update_promise->get_future();
```

Comment:
Is all this promise stuff just used for the delay?

Reply:
Yea the promise+future is used just to verify that the callback is called.
The promise is used to set a value, and the corresponding future is used to read the value once it has been set.